wordpress-mobile/WordPress-Android
# Allow a post to be marked as sticky

Fixes #7953

**This is still in progress, please don't merge**

### Description
We wanna allow the users to mark a post as sticky from the Android app. This is already present in iOS and Web.
A blocker for this PR is the approval of https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2124 and an update of the FluxC library (which will add the "sticky" field to the `PostModel`).

### Work done

A switch was added to Post Settings screen. This is loaded and updated in the `PostModel` (from FluxC), which syncs it to the Wordpress servers.
This switch only appears for Posts, so it is hidden for Pages.

### Screenshot
<img src="https://user-images.githubusercontent.com/628412/133897108-4f0a4a41-8911-4300-9e73-507567f71b8a.jpeg" width=350 />

### To be done

* Identify any regression areas
* Localize strings
* Add details to the release-notes

To test:
1. Open a site with posts in it
2. Open a post to edit it
3. Tap to edit "Post settings"
4. Toggle the switch "Stick post to the front page"
5. Update the post with the changes (sync it)
6. Check on Web or iOS that the changes were applied
7. Revert the change on Web or iOS
8. Check on Android that the change is reflected there

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests

3. What automated tests I added (or what prevented me from doing so)
There are no tests surrounding that Fragment. And any unit tests added will be testing the FluxC library.

PR submission checklist:
